code/modules/mob/living/simple_animal/bot/medbot.dm

```diff
@@ -588,6 +595,76 @@
 	var/area/location = get_area(src)
 	speak("Medical emergency! [crit_patient || "A patient"] is in critical condition at [location]!", radio_channel)
 
+/mob/living/simple_animal/bot/medbot/cola
+	name = "Sponsored Medibot"
+	desc = "Even medical isn't safe from corporate greed. May contain cola."
+	radio_key = /obj/item/encryptionkey/medical_sponsor  // grants cargo
+	COOLDOWN_STATIC_DECLARE(fund)
+	var/static/bot_budget = 1000  // initial value (will get refilled instantly)
+	var/static/bot_budget_refill_size = 1000
+	var/cola_cost = 250  // at start, uses entire budget in 8 heals
```

Comment:
This isn't entirely true, I don't think cargo starts off with any money at all atm, and I don't think we should have such a comment anyways since it'll be very confusing if economy changes.

Reply:
it uses a local budget that starts at 1000

Reply:
You're right, I updated my main comment with my second thoughts on it, but I think it would be better to use ``/datum/component/reagent_refiller``, which is what we use for service borgs, to handle this instead. It *might* need minor rewording but from looking at the component, it looks perfect for what you want to do here.

Reply:
alright so im a bit confused here. should i subclass reagent_refiller in order to make it work without a reagent object, or make a new datum based on it? or what? the intended effect is every 2 minutes from the bot's creation, it gets another 1000 credits to "give" to cargo. is there any kinda component that just runs every 2 minutes to fill a certain var?

Reply:
You can make medbots have a reagent contained built-in (or just the cola subtype, to be honest) and have the component added to that on Initialize after adding it, or maybe you can get away with having the medbot be the reagents holder and remove the checks for such in reagent_refiller component, since only carbons+ should metabolize reagents (though I don't think it's a great idea).

Reply:
ohh, i get it ok

Reply:
nevermind i do not get it help

Reply:
Like how a beepsky has a baton as a var which they use to attack people, you can have a reagent container on a medibot to hold the reagents you want them to inject. You can put the var on the cola bot, and new() it on Initialize, where you'll also add the self-refilling component.